src/Catalog/Dnx/DnxMaker.cs

```diff
@@ -177,12 +221,24 @@ private StorageContent CreateContent(IEnumerable<string> versions)
 
         private static string GetRelativeAddressNuspec(string id, string version)
         {
-            return $"{NuGetVersion.Parse(version).ToNormalizedString()}/{id}.nuspec"; 
+            return $"{NuGetVersion.Parse(version).ToNormalizedString()}/{id}.nuspec";
         }
 
         public static string GetRelativeAddressNupkg(string id, string version)
         {
-            return $"{NuGetVersion.Parse(version).ToNormalizedString()}/{id}.{NuGetVersion.Parse(version).ToNormalizedString()}.nupkg";
+            if (string.IsNullOrEmpty(id))
+            {
+                throw new ArgumentException(Strings.ArgumentMustNotBeNullOrEmpty, nameof(id));
+            }
+
+            if (string.IsNullOrEmpty(version))
+            {
+                throw new ArgumentException(Strings.ArgumentMustNotBeNullOrEmpty, nameof(version));
+            }
+
+            var normalizedVersion = NuGetVersion.Parse(version).ToNormalizedString();
```

Comment:
>NuGetVersion.Parse(version).ToNormalizedString() [](start = 36, length = 48)

You could use `NuGetVersionUtility.NormalizeVersion` here

Reply:
That would introduce undesirable leniency in this API.  Currently, if you call this method with an unparseable version:

    DnxMaker.GetRelativeAddressNupkg(id: "a", version: "b")

...it will throw with `System.ArgumentException: 'b' is not a valid version string.`  I think that is correct behavior.  If `GetRelativeAddressNupkg(...)` instead used `NuGetVersionUtility.NormalizeVersion(...)`, the above method call would not throw and would return:

    b/a.b.nupkg

The previous implementation threw on invalid version and I'm preserving that behavior.